test/integration/lib/testPrepper.js

```diff
@@ -9,46 +10,47 @@ var config = new Config('./test/integration/lib/config/foxbox.js');
 
 var testPrepper = (function() {
 
+  function expectLightIsOn(response) {
+   expect(response).to.have.status(200);
+   expect(response.body.result).equals('success');
+   expect(philipshue_server.lastCmd()).to.contain('"on":true');
+  }
+
+  function turnOnLight(lightinfo, serviceURL, index) {
+    return chakram.put(serviceURL + lightinfo[index].id + '/state', {'on': true});
+  }
+
   function beforeTest(done) { 
-    // start nupnp server
+
+    console.log("beforetest start");
     nupnp_server.start(config.get('nupnp_server.id'),
       config.get('philips_hue.url') +':'+config.get('philips_hue.port'),
       config.get('nupnp_server.port'));
     
-    // configure philips hub simulator
     philipshue_server.setup(config.get('philips_hue.port'));
 
-    // start foxbox
     foxboxManager.fullOptionStart(done);        
   }
 
   function afterTest(done) { 
-    var killCount = 0;
-
-    var getKill = function(){
-      killCount++;
-      if(killCount === 2){
-        done();
-      }
-    };
-    nupnp_server.stop(getKill());
-    philipshue_server.stop(getKill());
-    foxboxManager.removeUsersDB();
+    
     foxboxManager.killFoxBox();
+    nupnp_server.stop();
+    philipshue_server.stop();
+    foxboxManager.removeUsersDB();
+    console.log("aftertest done");
```

Comment:
This is where I'm having a problem.  This console.log gets executed after the console.log in before() (line 25 of this file).  How can I make sure that it waits sufficiently? @JohanLorenzo